### PR TITLE
SEAB-6187: Adjust messaging for notebooks

### PR DESCRIPTION
**Pull request by svonworl**
**Description**
This PR adjusts the text of various UI components to mention and/or make sense for notebooks.  Specifically, it addresses the following issues:

* Notebook version “Info” dialog displays “Test Parameter File(s)”, should be “Test Files”.
* Advanced Search dialog box should include “notebooks” alongside “tools and workflows”.
* “Snapshot version” dialog refers to the notebook as a “workflow”.
* “Request DOI” dialog refers to the notebook as a “workflow”.
* “Export to ORCID” dialog refers to the notebook as a “workflow”.
* “Delete notebook” dialog refers to a “checked or checker workflow”, which doesn’t make sense and should be removed for notebooks.

For more details, see this doc: https://docs.google.com/document/d/1s3OU1kDcVwjPIOdgsmmaJTKGD2ZrojgZS5VYUt8njf8

**Review Instructions**
For each of the issues in the above list, navigate to the affected component and confirm that the fix was made correctly.  If this PR introduced a change that allowed the component's text to vary by entry type, also navigate to the same component in the context of a workflow, and confirm that nothing was broken. 

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-6187

**Security**
No concerns.

- [x] Check that your code compiles by running `npm run build`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.
- [x] Check whether this PR disables tests. If it legitimately needs to disable a test, create a new ticket to re-enable it in a specific milestone.

### CODE REVIEW

**Review by coverbeck — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]

File: src/app/entry/delete/dialog/delete-entry-dialog.component.html [hunk 1]
```diff
@@ -7,9 +7,9 @@ <h1 mat-dialog-title>Are you sure you want to delete this Dockstore {{ term }}?<
   </p>
   <p>A {{ term }} can be deleted if it:</p>
   <ul>
-    <li>Has never been published,</li>
```
Comment (denis-yuen) [inline src/app/entry/delete/dialog/delete-entry-dialog.component.html:10]: I mean, if we're fixing punctuation, these don't need periods or commas
See the "Research office" example of https://www.imperial.ac.uk/brand-style-guide/writing/grammar/punctuation/bullet-points/

File: src/app/entry/delete/dialog/delete-entry-dialog.component.html [hunk 2]
```diff
@@ -7,9 +7,9 @@ <h1 mat-dialog-title>Are you sure you want to delete this Dockstore {{ term }}?<
...
-    <li>Was created on/after April 1, 2022, and</li>
-    <li>Is not a checked or checker workflow.</li>
```
Comment (denis-yuen) [inline src/app/entry/delete/dialog/delete-entry-dialog.component.html:12]: Is the condition right here? 
Should this be displayed for a tool which itself is not a workflow?

**Review by denis-yuen — COMMENTED** [review]
(more for the condition question)

File: src/app/entry/delete/dialog/delete-entry-dialog.component.html (hunk 2 above)
Comment (svonworl) [reply src/app/entry/delete/dialog/delete-entry-dialog.component.html:12]: Yeah, it's wrong, I didn't realize tools could be checked.  Maybe I'll change it to not display for services or notebooks.  I probably should change the last item to say "Is not a checked workflow/tool or checker workflow."  That's kind of a mouthful, maybe "Is not checked or a checker workflow."?

Comment (svonworl) [reply src/app/entry/delete/dialog/delete-entry-dialog.component.html:12]: Done.

File: src/app/entry/delete/dialog/delete-entry-dialog.component.html (hunk 1 above)
Comment (svonworl) [reply src/app/entry/delete/dialog/delete-entry-dialog.component.html:10]: The bullet items "complete the introductory stem", thus each forming full sentences, so according to this, at least, they should be punctuated:
https://www.microsoft.com/en-us/microsoft-365-life-hacks/writing/punctuating-bullet-point-lists

Comment (denis-yuen) [reply src/app/entry/delete/dialog/delete-entry-dialog.component.html:10]: If you're going with that guide, the first word of each shouldn't be capitalized then

**Review by denis-yuen — APPROVED** [review]

**Pull request merged**